src/Gateway/PayplugGateway.php

```diff
@@ -288,6 +290,16 @@ public function validate_payment($id = null, $save_request = true, $ipn = false)
 			$order_id = (int) $id;
 		}
 
+		if (empty($order_id)) {
+			$order_id = apply_filters(
+				'woocommerce_thankyou_order_id',
+				absint($wp->query_vars['order-received'])
+			);
+			if (empty($order_id)) {
+				return;
+			}
+		}
+
 		if (empty($order_id)) {
```

Comment:
this will never happen, amd I right?
the previous code, weither return empty or returns an order_id.
Can we remove it?

Reply:
or if the filter doesn't return an order_id you wanna rty to get it from the `key`?

Reply:
you are right, didn't pay attention, pushed a fix already